src/python/flowpy/python_field_base.py

```diff
@@ -0,0 +1,81 @@
+#!/bin/python3
+# author: David Flanderka
+
+import sys
+#import flowpy
+import numpy as np
+
+class PythonFieldBase():
+    _instances = dict()
+
+    @staticmethod
+    def create(module, class_name):
+        """ Creates instance of class_name if doesn't exist. Stores its to _instances and returns. """
+        if class_name not in PythonFieldBase._instances:
+            # module = __import__(module_name)
+            class_ = getattr(module, class_name)
+            PythonFieldBase._instances[class_name] = class_()
+        return PythonFieldBase._instances.get(class_name, None)
+
+
+    def __init__(self):
+        """ Initialize object and its data members """
+        self.region_chunk_begin = 0
+        self.region_chunk_end = 300
+        self.t = 0.0
+        self.used_fields_dict = dict()
+        self.result_fields_dict = dict()
+
+
+    def __getattr__(self, attr):
+        """ Allows direct access to items in 'used_fields_dict' field data dictionary.
+            Example: use 'self.field_name' instead of 'self.used_fields_dict["field_name"]' """
+        # print(attr)
+        cache_data = self.used_fields_dict.get(attr, None)
+        return cache_data[..., self.region_chunk_begin:self.region_chunk_end]
+        
+
+    def used_fields(self):
+        """ Method allows to define list of fields used in evaluation. This is the default defintion
```

Comment:
Remove this method, and introduce a key 'used_fields' in the FieldPython record.